### PR TITLE
bam to counts conversion 

### PR DESCRIPTION
**Pull request by cansavvy**
# Description

 

This is adding the bam to counts conversion that's done by the custom scripts [counter_efficient.R](https://github.com/FredHutch/pgMAP_pipeline/blob/main/workflow/scripts/counter_efficient.R)and [combine_counts.py](https://github.com/FredHutch/pgMAP_pipeline/blob/main/workflow/scripts/combine_counts.py)  in the original pipeline. 

<del>The only thing I find confusing is that the counts I'm getting (after calculating weights) do not turn out as whole numbers. And I'm not sure why they do in the original data. <\del>

Figured it out! See below. It's lining up with the expected counts from the original pipeline configuration. 🎉 

## Type of change

 

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

 

I haven't developed unit tests for this however the examples documented in the code are working examples and are how I have tested these functions. 

# Checklist:

- [x] My code follows the style guidelines of this project 
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by cansavvy** [comment]
Screenshot 2024-10-24 at 3 20 56 PM

I figured it out. Here's a plot to prove it. I'm putting proof of concept plots in the inst/extdata/rmd folder here so we have them for our records.

File: DESCRIPTION [hunk 1]
```diff
@@ -1,22 +1,14 @@
 Type: Package
-Package: gimap
-Title: Calculate Genetic Interactions for CRISPR Targets
+Package: pgmap
```
Comment (cansavvy) [inline DESCRIPTION:2]: Also did some package docs updates

**Comment by marissafujimoto** [comment]
@cansavvy Looks great overall! Much cleaner compared to the original pipeline! Ship it. 

My biggest thought when reading this (and I think it's out of scope for this PR) is whether we can do this count without loading the bams into memory. Instead maybe iteratively building up the counts and using O(# paired guides) memory by streaming the bam files.

Another thought I had was whether the bam / alignment step is necessary given that mageck argues that you should go directly from fastq and not count partial alignments, though I'm not sure if this should be different for paired guides or how much of our example data is perfectly aligned: https://sourceforge.net/p/mageck/wiki/advanced_tutorial/#tutorial-1-allow-mismatches-for-read-mapping

**Comment by cansavvy** [comment]
> @cansavvy Looks great overall! Much cleaner compared to the original pipeline! Ship it. 

I'll ship it after I make the tests pass. 👍 



> My biggest thought when reading this (and I think it's out of scope for this PR) is whether we can do this count without loading the bams into memory. Instead maybe iteratively building up the counts and using O(# paired guides) memory by streaming the bam files.
> 

I ran this locally without a problem but certainly saving compute time and memory is good. And we should determine how big are the biggest files these assays might have to think more specifically about bench marking. 


> 
> Another thought I had was whether the bam / alignment step is necessary given that mageck argues that you should go directly from fastq and not count partial alignments, though I'm not sure if this should be different for paired guides or how much of our example data is perfectly aligned: https://sourceforge.net/p/mageck/wiki/advanced_tutorial/#tutorial-1-allow-mismatches-for-read-mapping


Interesting thanks for sharing! I'll dig into this tomorrow!

**Comment by cansavvy** [comment]
Still some troubles with the Rsamtools dependency but I'll move on

**Pull request merged**